docs/snapshotters/erofs.md

```diff
@@ -0,0 +1,143 @@
+I believe there are clear differences if looking into `s.mount()` and it highly
+tightens to the EROFS internals.  Currently, it's not quite clear to form an
+enhanced OverlayFS snapshotter directly, and (I think) it's not urgent since
+in the very beginning, it'd be better to be left as an independent snapshotter
+so that existing overlayfs users won't be impacted by the new behaviors and
+users could have a chance to try and develop the related ecosystems (such as
+ComposeFS, confidential containers, gVisor, Kata, gVisor, and more) together.
+
+## Use Cases
+
+The EROFS snapshotter can benefit to several use cases:
+
+For runC containers, instead of unpacking individual files into a directory
+on the backing filesystem, it applies OCI layers into EROFS blobs, therefore:
+
+ - Improved image unpacking performance (~14% for WordPress image with the
+   latest erofs-utils 1.8.2) due to reduced metadata overhead;
+
+ - Full data protection for each snapshot using the S_IMMUTABLE file attribute
+   or fsverity. Currently, fsverity can only protect blob data in the content
+   store;
+
+ - Parallel unpacking can be supported in a more reliable way (fsync) compared
+   to the overlayfs snapshotter (syncfs);
+
+ - Native EROFS layers can be pulled from registries without conversion.
+
+For VM containers, the EROFS snapshotter can efficiently pass through and share
+image layers, offering several advantages (e.g. better performance and smaller
+memory footprints) over [virtiofs](https://virtio-fs.gitlab.io) or
+[9p](https://www.kernel.org/doc/Documentation/filesystems/9p.txt).  Besides,
+the popular application kernel [gVisor](https://gvisor.dev/) also supports
+[EROFS](https://github.com/google/gvisor/pull/9486) for efficient image
+pass-through.
+
+## Usage
+
+### Checking if the EROFS snapshotter and differ are available
+
+To check if the EROFS snapshotter is available, run the following command:
+
+```bash
+$ ctr plugins ls | grep erofs
+```
+
+The following message will be shown like below:
+```
+io.containerd.snapshotter.v1           erofs                    linux/amd64    ok
+io.containerd.differ.v1                erofs                    linux/amd64    ok
+```
+
+### Ensure that EROFS is available
+
+On newer Ubuntu/Debian systems, it can be installed directly using the apt
+command, and on Fedora it can be installed directly using the dnf command.
+
+# Debian/Ubuntu
+$ apt install erofs-utils
+# Fedora
+$ dnf install erofs-utils
+
+Make sure that erofs-utils version is 1.7 or higher.
+
+Before using EROFS snapshotter, also make sure the _EROFS kernel module_ is
+loaded: it can be loaded with `modprobe erofs`.
+
+### Configuration
+
+The following configuration can be used in your containerd `config.toml`. Don't
+forget to restart containerd after changing the configuration.
+
+```
+  [plugins."io.containerd.service.v1.diff-service"]
+    default = ["erofs","walking"]
```

Comment:
I'm seeing this warning in the logs:
```
WARN[2025-01-31T14:27:36.659539529+02:00] multiple differs match for platform, set `differ` option to choose, skipping "walking" 
```

Reply:
it seems it relates to transfer service, but I don't really know the details too.
EROFS snapshotter can only work effectively with the related differ, so just
leaving "default = ["erofs"]" is fine in practice.